components/formatted_markdown_message.jsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedHTMLMessage, injectIntl, intlShape} from 'react-intl';
+import PropTypes from 'prop-types';
+import marked from 'marked';
+
+const UNESCAPED_TAGS = ['BR'];
+const TARGET_BLANK_URL_PREFIX = '!';
+
+class FormattedMarkdownMessage extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.renderer = new marked.Renderer();
+        this.renderer.link = FormattedMarkdownMessage.linkF;
```

Comment:
Maybe make a new class that extends `marked.Renderer` to override the link method? I think that might be more the more canonical way to do this

Reply:
Do you have a reference for that proposal? The solution see here is base off of their documentation: https://github.com/markedjs/marked/blob/master/docs/USING_PRO.md#the-renderer

Reply:
Their solution isn't using ES6 features like we have access to. If you look at [the main renderer](https://github.com/mattermost/mattermost-webapp/blob/master/utils/markdown/renderer.jsx#L11), we make a subclass of `marked.Renderer` and override the methods there instead of the older JS style of replacing the method in each instance of the class.